web/src/features/charts/FuturePrice.tsx

```diff
@@ -81,31 +81,33 @@ export function FuturePrice({ futurePrice }: { futurePrice: FuturePriceData | nu
                         : ''
                     }
                   >
-                    {dateIsFirstHourOfTomorrow(new Date(date)) && (
-                      <div className="flex flex-col py-1" data-test-id="tomorrow-label">
-                        <HorizontalDivider />
-                        <TommorowLabel date={date} t={t} i18n={i18n} />
-                      </div>
-                    )}
-                    <div className="flex flex-row justify-items-end gap-2 px-1">
-                      <TimeDisplay date={date} granularity={granularity} />
-                      <PriceDisplay
-                        price={price}
-                        currency={futurePrice.currency}
-                        i18n={i18n}
-                      />
-                      <div className="flex h-full w-full flex-row self-center">
-                        {hasNegativePrice && (
-                          <div
-                            className="flex flex-row justify-end"
-                            style={{
-                              width: `${negativePercentage}%`,
-                            }}
-                            data-test-id="negative-price"
-                          >
-                            {price < 0 && (
+                    <div>
+                      {dateIsFirstHourOfTomorrow(new Date(date)) && (
+                        <div className="flex flex-col py-1" data-test-id="tomorrow-label">
+                          <HorizontalDivider />
+                          <TommorowLabel date={date} t={t} i18n={i18n} />
+                        </div>
+                      )}
+                      <div className="flex flex-row justify-items-end gap-2 px-1">
+                        <TimeDisplay date={date} granularity={granularity} />
+                        <PriceDisplay
+                          price={price}
+                          currency={futurePrice.currency}
+                          i18n={i18n}
+                        />
+                        <div className="flex h-full w-full flex-row self-center">
+                          {hasNegativePrice && price < 0 && (
+                            <div
+                              className="flex flex-row justify-end"
+                              style={{
+                                width: `calc(calc(calc(100% - 12px) * ${
+                                  negativePercentage / 100
+                                }) + 12px)`,
+                              }}
```

Comment:
```suggestion
                                                            style={{
                                width: `calc(100% * ${negativePercentage / 100} + 12px - 12px * ${negativePercentage / 100})`,
                              }}
```
You should be able to use just one calc. Making this even simpler! But very nice approach!